maat-court-data-api/src/main/java/gov/uk/courtdata/authorization/service/AuthorizationService.java

```diff
@@ -0,0 +1,28 @@
+package gov.uk.courtdata.authorization.service;
+
+import gov.uk.courtdata.exception.ValidationException;
+import gov.uk.courtdata.repository.AuthorizationRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class AuthorizationService {
+    private final AuthorizationRepository authorizationRepository;
+
+    public boolean isRoleActionAuthorized(String username, String action) {
+        List<String> validActions = authorizationRepository.getAvailableActions();
```

Comment:
@Mhowell494  This could potentially be an expensive operation i.e. returning all actions from the DB everytime there is a authorization request. Please could you consider rewriting this to use a single query and then check if the returned list is empty or contains the requested action ?